quarkus-opentelemetry-exporter-gcp/runtime/src/main/java/io/quarkiverse/opentelemetry/exporter/gcp/runtime/GcpExporterConfig.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkiverse.opentelemetry.exporter.gcp.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+public class GcpExporterConfig {
+    @ConfigRoot(name = "opentelemetry.tracer.exporter.gcp", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+    public static class GcpExporterBuildConfig {
+        /**
+         * GCP SpanExporter support.
+         * <p>
+         * GCP SpanExporter support is enabled by default.
+         */
+        @ConfigItem(defaultValue = "true")
+        public Boolean enabled;
+    }
+
+    @ConfigRoot(name = "opentelemetry.tracer.exporter.gcp", phase = ConfigPhase.RUN_TIME)
+    public static class GcpExporterRuntimeConfig {
+        /**
+         * For Cloud Run environments we cannot use BatchSpanProcessor. Cloud Run doesn't support background processes.
+         * Reference: https://cloud.google.com/trace/docs/setup/java-ot#export
+         */
+        @ConfigItem(defaultValue = "false")
+        public Boolean cloudrun;
```

Comment:
Not sure where you are using this

Reply:
According to https://cloud.google.com/trace/docs/setup/java-ot#export - `BatchSpanProcessor` cannot be used in Cloud Run applications.

Something along those lines:
```
if (!cloudrun) { 
   return BatchSpanProcessor(); 
} else {
   return SimpleSpanProcessor();
}
```

I'm not sure how to correctly do that.

Reply:
You can produce one processor or the other, according to the flag.

Reply:
You can also produce both and decide in the recorder according to the config.

Reply:
I've opted out to produce both and decide in the recorder.